avocado/utils/linux_modules.py

```diff
@@ -208,34 +208,91 @@ def get_loaded_modules():
         return [astring.to_text(_.split(b' ', 1)[0]) for _ in proc_modules]
 
 
-def check_kernel_config(config_name):
+def check_kernel_config(config_name, session=None):
     """
     Reports the configuration of $config_name of the current kernel
 
     :param config_name: Name of kernel config to search
+    :param session: guest session, command is run on host if None
     :type config_name: str
     :return: Config status in running kernel (NOT_SET, BUILTIN, MODULE)
     :rtype: :class:`ModuleConfig`
     """
+    if session is None:
+        kernel_version = platform.uname()[2]
+        config_file = '/boot/config-' + kernel_version
+        # Sometimes a kernel config's name may contain another kernel config
+        # E.g.
+        # CONFIG_PREEMPT_VOLUNTARY=y
+        # CONFIG_PREEMPT is not set
+        # So we need to search CONFIG_XXX= in kconfig file
+        config_info = process.run('grep {}= {}'.format(config_name,
+                                  config_file), shell=True)\
+            .stdout_text.strip()
+    else:
+        kernel_version = session.cmd('uname -r').strip()
+        config_file = '/boot/config-' + kernel_version
+        config_info = session.cmd('grep {}= {}'.format(config_name,
+                                  config_file)).strip()
+
+    line = config_info.split('=')
+    if len(line) != 2:
+        return ModuleConfig.NOT_SET
+
+    config = line[0].strip()
+    LOG.debug("Get config {}, target is {}".format(config, config_name))
+    if config == config_name:
+        option = line[1].strip()
+        if option == "m":
+            return ModuleConfig.MODULE
+        else:
+            return ModuleConfig.BUILTIN
+    return ModuleConfig.NOT_SET
 
-    kernel_version = platform.uname()[2]
 
-    config_file = '/boot/config-' + kernel_version
-    with open(config_file, 'r') as kernel_config:  # pylint: disable=W1514
-        for line in kernel_config:
-            line = line.split('=')
+def kconfig_is_builtin(config_name, session=None):
+    """
+    Return True is kernel config is BUILTIN
 
-            if len(line) != 2:
-                continue
+    :param config_name: Name of kernel config to check
+    :param session: Guest session, command is run on host if None
 
-            config = line[0].strip()
-            if config == config_name:
-                option = line[1].strip()
-                if option == "m":
-                    return ModuleConfig.MODULE
-                else:
-                    return ModuleConfig.BUILTIN
-    return ModuleConfig.NOT_SET
+    :return: Return True if kconfig is BUILTIN, otherwise False.
+    """
+    if check_kernel_config(config_name, session) == ModuleConfig.BUILTIN:
```

Comment:
A shorthand, as a suggestion, that is also very Pythonic is:

```python
return check_kernel_config(config_name, session) is ModuleConfig.BUILTIN
```

Reply:
> A shorthand, as a suggestion, that is also very Pythonic is:
> 
> ```python
> return check_kernel_config(config_name, session) is ModuleConfig.BUILTIN
> ```
> 
> With that being said, I am curious about your use case and the real need for these 3 new functions. Wouldn't the shorthand have about the same readability of the newly proposed functions?

Yes. The 3 new functions could be replaced by the shorthand. 
I added the 3 new functions because the shorthand is too long.
We always need to split it into 2 line when we use it.

Reply:
> > A shorthand, as a suggestion, that is also very Pythonic is:
> > ```python
> > return check_kernel_config(config_name, session) is ModuleConfig.BUILTIN
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > With that being said, I am curious about your use case and the real need for these 3 new functions. Wouldn't the shorthand have about the same readability of the newly proposed functions?
> 
> Yes. The 3 new functions could be replaced by the shorthand. I added the 3 new functions because the shorthand is too long. We always need to split it into 2 line when we use it.

OK, my comment wasn't a hard NACK.  I can see these shorthand versions being helpful, and the multi-line nature is a valid argument IMO.